presto-main/src/main/java/io/prestosql/execution/buffer/BroadcastOutputBuffer.java

```diff
@@ -231,6 +237,11 @@ public void enqueue(List<SerializedPage> pages)
 
         // drop the initial reference
         serializedPageReferences.forEach(SerializedPageReference::dereferencePage);
+
+        // if the buffer is full for first time, update the task status
```

Comment:
please add some more rationale why we do this (for future readers)

Reply:
Added a better comment and also added a check in the condition for `canAddBuffers` (updated test too). request to please check it once as well